File: Labtwo/JS/pr3.js
function funcnum(numb) {
    // Проверка что все символы - цифры
    if (!/^\d+$/.test(numb)) {
        return;
    }
    
    // Проверка длины
    if (numb.length < 2) {
        return;
    }
    
    const num = parseInt(numb);
    let sum = 0;
    let proizv = 1;
    
    process.stdout.write(`${num}: `);
    
    // Сумма цифр
    for (let i = 0; i < numb.length; i++) {
        const d = parseInt(numb[i]);
        sum += d;
        process.stdout.write(numb[i]);
        if (i < numb.length - 1) {
            process.stdout.write("+");
        }
    }
    process.stdout.write(`=${sum}, `);
    
    // Произведение цифр
    for (let i = 0; i < numb.length; i++) {
        const d = parseInt(numb[i]);
        proizv *= d;
        process.stdout.write(numb[i]);
        if (i < numb.length - 1) {
            process.stdout.write("*");
        }
    }
    process.stdout.write(`=${proizv}, `);
    
    // Модуль разности
    const diff = Math.abs(sum - proizv);
    console.log(`|${sum}-${proizv}|=${diff}`);
}

const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Write your number: ', input => {
    input.split(' ').forEach(numb => {
        funcnum(numb);
    });
    readline.close();
});

File: Labtwo/JAVA/pr3.java
import java.util.Scanner;

public class NumberProcessor {
    /**
     * Метод для обработки числовой строки
     * @param numb - строка, содержащая число
     */
    public static void processNumber(String numb) {
        // Проверка что все символы в строке - цифры
        for (char c : numb.toCharArray()) {
            if (!Character.isDigit(c)) {
                return; // Если найден нецифровой символ, выходим из метода
            }
        }

        // Проверка что число содержит минимум 2 цифры
        if (numb.length() < 2) {
            return;
        }

        // Преобразование строки в целое число
        int num = Integer.parseInt(numb);
        int sum = 0;       // Сумма цифр
        int product = 1;   // Произведение цифр

        // Вывод исходного числа
        System.out.print(num + ": ");

        // Вычисление и вывод суммы цифр
        for (int i = 0; i < numb.length(); i++) {
            int digit = Character.getNumericValue(numb.charAt(i));
            sum += digit;
            System.out.print(digit);
            if (i < numb.length() - 1) {
                System.out.print("+"); // Вывод плюсов между цифрами
            }
        }
        System.out.print("=" + sum + ", ");

        // Вычисление и вывод произведения цифр
        for (int i = 0; i < numb.length(); i++) {
            int digit = Character.getNumericValue(numb.charAt(i));
            product *= digit;
            System.out.print(digit);
            if (i < numb.length() - 1) {
                System.out.print("*"); // Вывод звездочек между цифрами
            }
        }
        System.out.print("=" + product + ", ");

        // Вычисление и вывод модуля разности
        int diff = Math.abs(sum - product);
        System.out.println("|" + sum + "-" + product + "|=" + diff);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Write your numbers separated by spaces:");
        String input = scanner.nextLine();

        // Разделение введенной строки по пробелам
        String[] numbers = input.split("\\s+");
        
        // Обработка каждого числа
        for (String num : numbers) {
            processNumber(num);
        }
        
        scanner.close();
    }
}